crates/ty_python_semantic/src/types/constraints.rs

```diff
@@ -207,7 +209,7 @@ impl<'db> ConstraintSet<'db> {
                 lower.top_materialization(db),
                 upper.bottom_materialization(db),
             ),
-            TypeRelation::Assignability => (
+            TypeRelation::Assignability | TypeRelation::ConstraintSetAssignability => (
                 lower.bottom_materialization(db),
                 upper.top_materialization(db),
```

Comment:
Is this where the wrong bottom materializations are coming from?

I think in the ecosystem I also saw some cases of top materializations inferred where they shouldn't be...